native/src/hooks/useTtsPlayer.ts

```diff
@@ -1,23 +1,35 @@
-import { useContext, useEffect } from 'react'
+import { useContext, useEffect, useMemo } from 'react'
+import segment from 'sentencex'
 
-import { ttsContext, TtsContextType } from '../components/TtsPlayer'
+import { parseHTML } from 'shared'
+import { EventModel, LocalNewsModel, PageModel, TunewsModel } from 'shared/api'
 
-const useTtsPlayer = (content?: string, title?: string): TtsContextType => {
+import { ttsContext, TtsContextType } from '../components/TtsContainer'
+
+const useTtsPlayer = (model?: PageModel | LocalNewsModel | TunewsModel | EventModel | undefined): TtsContextType => {
   const tts = useContext(ttsContext)
+  const sentences = useMemo(() => {
+    if (model) {
+      const content = parseHTML(model.content)
+      return [model.title, ...segment(tts.languageCode, content)]
+    }
+
+    return null
+  }, [model, tts.languageCode])
 
   useEffect(() => {
-    if (content && title) {
-      tts.setContent(content)
-      tts.setTitle(title)
+    if (sentences) {
+      tts.setSentences(sentences)
+      tts.setTitle(model?.title ?? '')
     }
     return () => {
-      tts.setContent(null)
+      tts.setSentences([])
+      tts.setTitle('')
     }
-  }, [content, title, tts])
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
tts is an object that changes every time so its wrong to add it as dependency.. any suggestions?

Reply:
It shouldn't change on every render as you wrap it in a `useMemo` and all of its dependencies are from `useState`. However, the feeling that it changes too often might be because you also expose `sentenceIndex` which changes frequently. Since its not used anywhere, you can just remove that though.

Reply:
Tried to put setSentenceIndex  in a callback function at TtsContainer but still showing:
Warning: Maximum update depth exceeded. This can happen when a component calls setState inside useEffect, but useEffect either doesn't have a dependency array, or one of the dependencies changes on every render.